From 7ee86be887cfedbfded33f6dcab481c950d9d620 Mon Sep 17 00:00:00 2001
From: Todd Kjos <tkjos@google.com>
Date: Fri, 25 Oct 2019 11:39:57 -0700
Subject: [PATCH 576/830] ANDROID: init: GKI: enable hidden configs for media
X-NVConfidentiality: public

Add hidden configs to GKI_HACKS_to_fIX so they are
enabled for loadable media modules build out-of-tree

Bug: 142892658
Test: Built for arm64/x86_64. Boot on cuttlefish
Change-Id: I137a89f69822332324a1f4a3a1150be9cd66709d
Signed-off-by: Todd Kjos <tkjos@google.com>
[maennich: Folded c387eb07b04c ("ANDROID: fix VIDEOBUF2_CORE dependency
           in 'allmodconfig' builds") into this patch]
Signed-off-by: Matthias Maennich <maennich@google.com>
---
 arch/arm64/configs/gki_defconfig | 1 -
 arch/x86/configs/gki_defconfig   | 1 -
 init/Kconfig.gki                 | 9 +++++++++
 3 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/configs/gki_defconfig b/arch/arm64/configs/gki_defconfig
index 33b742543ec4..e2b7f40cb55a 100644
--- a/arch/arm64/configs/gki_defconfig
+++ b/arch/arm64/configs/gki_defconfig
@@ -296,7 +296,6 @@ CONFIG_WATCHDOG=y
 CONFIG_MFD_ACT8945A=y
 CONFIG_MFD_SYSCON=y
 CONFIG_REGULATOR=y
-CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_MEDIA_CONTROLLER=y
 # CONFIG_VGA_ARB is not set
diff --git a/arch/x86/configs/gki_defconfig b/arch/x86/configs/gki_defconfig
index e19e3343d322..09dc0adca00d 100644
--- a/arch/x86/configs/gki_defconfig
+++ b/arch/x86/configs/gki_defconfig
@@ -256,7 +256,6 @@ CONFIG_GPIOLIB=y
 CONFIG_DEVFREQ_THERMAL=y
 # CONFIG_X86_PKG_TEMP_THERMAL is not set
 CONFIG_REGULATOR=y
-CONFIG_MEDIA_SUPPORT=y
 CONFIG_MEDIA_CAMERA_SUPPORT=y
 CONFIG_DRM=y
 # CONFIG_DRM_FBDEV_EMULATION is not set
diff --git a/init/Kconfig.gki b/init/Kconfig.gki
index 4db995c2fe92..c5a080e7784b 100644
--- a/init/Kconfig.gki
+++ b/init/Kconfig.gki
@@ -53,6 +53,14 @@ config GKI_HIDDEN_QCOM_CONFIGS
 	  These are normally selected implicitely when a module
 	  that relies on it is configured.
 
+config GKI_HIDDEN_MEDIA_CONFIGS
+	bool "Hidden Media configs needed for GKI"
+	select VIDEOBUF2_CORE
+	select MEDIA_SUPPORT
+	help
+	  Dummy config option used to enable hidden media configs.
+	  These are normally selected implicitely when a module
+	  that relies on it is configured.
 
 # LEGACY_WEXT_ALLCONFIG Discussed upstream, soundly rejected as a unique
 # problem for GKI to solve.  It should be noted that these extensions are
@@ -89,6 +97,7 @@ config GKI_HACKS_TO_FIX
 	select GKI_HIDDEN_GPIO_CONFIGS
 	select GKI_HIDDEN_QCOM_CONFIGS
 	select GKI_LEGACY_WEXT_ALLCONFIG
+	select GKI_HIDDEN_MEDIA_CONFIGS
 	help
 	  Dummy config option used to enable core functionality used by
 	  modules that may not be selectable in this config.
-- 
2.17.1

